docs/dedicated-data-endpoints/dedicated-data-endpoints.md

```diff
@@ -0,0 +1,114 @@
+
+<img src="./media/registry-private-link.png" width=500x>
+
+Unfortunately, VNet connectivity isn’t always an option.
+
+## Client Firewall Rules & Data Exfiltration Risks
+
+When connecting to a registry from on-prem hosts, IoT devices, custom build agents, or when Private Link simply is not an option, a client firewall rule could be applied, limiting access to specific resources.
+
+As customers lock down their clients and VNets, the wildcard opens a data exfiltration risk as rouge code running in the secured environment could write secure data out to other “bad-dev” storage accounts.
+
+<img src="./media/registry-data-exfiltration.png" width=500x>
+
+## Dedicated Data-endpoints
+
+Content is proxied through the registry service. Dedicated data-endpoints use the same proxy without requiring Private Links, providing known FQDNs for client firewall rules, avoiding the need for wildcards which blocks data exfiltration concerns. which blocks data exfiltration concerns.
+
+<img src="./media/registry-dedicated-data-endpoint.png" width=500x>
+
+## Enabling Dedicated Data-endpoints
+
+Switching to dedicated data-endpoints will impact clients that have configured firewall access to the existing `*.blob.core.windows.net` endpoints, causing pull failures. To assure clients have consistent access, add the new data-endpoints to the client firewall rules. Once completed, existing registries can enable dedicated data-endpoints through the az cli, or the Azure portal.
+
+### Private Preview Configuration
+
+Until the Portal and az cli are enabled, customers can use the `az rest` api to enable dedicated data-endpoints.
+
+- Set the registry default
+
+  ```sh
+  az configure --defaults acr=demo42
+  ```
+
+- Export the resource id of the registry
+
+  ```sh
+  export RESOURCE_ID=$(az acr show --query id -o tsv)
+  ```
+
+- Execute the REST api with the `az rest` command
+
+  ```sh
+  az rest --method patch --uri "$RESOURCE_ID?api-version=2019-12-01-preview" --body "{ \"properties\":{\"dataEndpointEnabled\":true}}" -o json
+  ```
+
+  Look for:
+
+  ```json
+  "dataEndpointEnabled": true,
+      "dataEndpointHostNames": [
+        "demo42.eastus.data.azurecr.io"
+  ```
+
+### az CLI
+
+Using [az cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) version 2.3.1 or greater, run the [az acr update](https://docs.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-update) command:
+
+```sh
+az acr update --name contoso --data-endpoint-enabled
+```
+
+To view the data-endpoints, including regional endpoints for geo-replicated registries, use the az acr show-endpoints cli:
+
+```sh
```

Comment:
@yugangw-msft is this released already. Did you review the API already?

Reply:
This was changed to reflect what @yugangw-msft provided